DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_bum_traffic_bridge_with_rif.py

```diff
@@ -0,0 +1,430 @@
+    """
+    Test Name: test_bridging_bum_traffic_bridge_with_rif
+    Test Suite: suite_functional_bridging
+    Test Overview: Verify forwarding/drop/trap of different broadcast, unknown-unicast
+                   and multicast traffic L2, IPv4 packet types within the same bridge domain.
+    Test Author: Kostiantyn Stavruk
+    Test Procedure:
+    1. Init bridge entity br0.
+    2. Set ports swp1, swp2, swp3, swp4 master br0.
+    3. Set entities swp1, swp2, swp3, swp4 UP state.
+    4. Set bridge br0 admin state UP.
+    5. Set ip address on bridge.
+    6. Get self MAC address on ingress port swp1.
+    7. Start tcpdump capture on DUT ingress port.
+    8. Send different types of packets from source TG.
+    9. Analyze counters: a) TX vs RX counters according to expected values;
+                         b) Trapped packets to CPU.
+    """
+
+    bridge = "br0"
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 2)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    traffic_duration = 10
+
+    out = await IpLink.add(
+        input_data=[{device_host_name: [
+            {"device": bridge, "type": "bridge"}]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that bridge created.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {"device": bridge, "operstate": "up"}]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that bridge set to 'UP' state.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "master": bridge, "operstate": "up"} for port in ports]}])
+    err_msg = f"Verify that bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await IpAddress.add(
+        input_data=[{device_host_name: [
+            {"dev": bridge, "prefix": "100.1.1.253/24"}]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Failed to add IP address to bridge.\n{out}"
+
+    out = await IpLink.show(input_data=[{device_host_name: [{"device": ports[0], "cmd_options": "-j"}]}],
+                            parse_output=True)
+    assert out[0][device_host_name]["rc"] == 0, f"Failed to display device attributes.\n{out}"
+
+    dev_attributes = out[0][device_host_name]["parsed_output"]
+    self_mac = dev_attributes[0]["address"]
+    srcMac = "00:00:AA:00:00:01"
+
+    address_map = (
+        # swp port, tg port,    tg ip,     gw,        plen
+        (ports[0], tg_ports[0], "1.1.1.2", "1.1.1.1", 24),
+        (ports[1], tg_ports[1], "1.1.1.3", "1.1.1.1", 24)
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
```

Comment:
Do we really need this step? Those IPs are not in use in the test case. Can we pass an empty config?